### PR TITLE
REL: Prepare for the NumPy 1.23.0rc1 release.

### PR DESCRIPTION
**Pull request by charris**
- Create 1.23.0-changelog.rst.
- Update 1.23.0-notes.rst (clears upcoming_changes).
- Update .mailmap.

### CODE REVIEW

File: doc/source/release/1.23.0-notes.rst [hunk 1]
```diff
@@ -4,42 +4,449 @@
...
+  (`gh-21029 <https://github.com/numpy/numpy/pull/21029>`__)
+
+* Changing to a dtype of different size in F-contiguous arrays is no longer
+  permitted. Deprecated since Numpy 1.11.0. See below for an extended
+  explanation of the effects of this change.
+
+  (`gh-20722 <https://github.com/numpy/numpy/pull/20722>`__)
+
+
+The behavior change associated with disallowing the dtype size changes in
```
Comment (charris) [inline doc/source/release/1.23.0-notes.rst:109]: Do we want to omit this? Maybe move it somewhere else?

**Comment by charris** [comment]
Feel free to suggest additional highlights.

File: doc/source/release/1.23.0-notes.rst (hunk 1 above)
Comment (seberg) [reply doc/source/release/1.23.0-notes.rst:109]: :+1: on simply removing the whole block and only keeping your nice bullet point.

**Comment by charris** [comment]
@mattip It looks like the wheel builder ran on this, although nothing was uploaded. Is that intended?

**Comment by mattip** [comment]
Adding "build" or "release" tags will trigger a wheel build run
https://github.com/numpy/numpy/blob/7e15fd77ca6c09989f219acf432c98a1036d14c5/.github/workflows/wheels.yml#L1-L14

**Comment by charris** [comment]
Hmm, could we make a more specific label for that? Maybe "build wheels".

**Comment by charris** [comment]
Or maybe just the one `[build wheel]` in the message. That could be added at any time during the PR process. Having too many ways to trigger builds is confusing. Also, needing to edit the tools to work in a fork seems unfortunate.

**Comment by charris** [comment]
Or maybe use a more specific label like `[test-wheels]` or maybe a `test-wheels` label.

**Comment by EwoutH** [comment]
I was working on some updates to the wheel CI, I opened a draft PR here: https://github.com/numpy/numpy/pull/21597. 

Could they still be included in 1.23.0rc1?

**Comment by charris** [comment]
> Could they still be included in 1.23.0rc1?

I would prefer to struggle through 1.23.0rc1 with what we have so it can be released today or tomorrow. If fixes are needed/useful we can backport them along the way.

**Pull request merged**